enhancements/cluster-logging/azure_workload_identity_auth.md

```diff
@@ -0,0 +1,264 @@
+  #[configurable(metadata(docs::examples = "/path/to/my/token"))]
+  pub federated_token_file_path: Option<String>,
+
+  /// The [Tenant ID][tenant_id] for authentication.
+  ///
+  /// [tenant_id]: https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-client-creds-grant-flow#third-case-access-token-request-with-a-federated-credential
+  #[configurable(metadata(docs::examples = "11111111-2222-3333-4444-555555555555"))]
+  pub tenant_id: Option<String>,
+
+  /// The [Client ID][client_id] for authentication.
+  ///
+  /// [client_id]: https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-client-creds-grant-flow#third-case-access-token-request-with-a-federated-credential
+  #[configurable(metadata(docs::examples = "11111111-2222-3333-4444-555555555555"))]
+  pub client_id: Option<String>,
+
+  /// The [Credential Kind][credential_kind] for authentication.
+  ///
+  /// [client_id]: https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-client-creds-grant-flow#third-case-access-token-request-with-a-federated-credential
+  #[configurable(metadata(docs::examples = "workloadIdentity"))]
+  pub credential_kind: String,
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+1. Relies on [this upstream vector PR](https://github.com/vectordotdev/vector/pull/22912) to implement the Azure Log Ingestion sink utilizing the Log Ingestion API.
+2. Current `master` branch of [upstream Vector](https://github.com/vectordotdev/vector), `>=v0.46.1`, as of 04/29/2025, utilizes `azure_identity@v0.21.0` which relies soley on environment variables for workload identity credentials and will not be sufficient when forwarding to multiple different Azure sinks. The PR above relies on `azure_identity@v0.21.0`.
+    - `azure_identity@v0.23.0` allows for setting `client_id`, `tenant_id`, etc. for authentication.
+    - [Workload Identity Credentials azure_identity@v0.23.0 SDK Ref](https://github.com/Azure/azure-sdk-for-rust/blob/azure_identity%400.23.0/sdk/identity/azure_identity/src/credentials/workload_identity_credentials.rs)
+3. The `customer_id` and `log_type` fields can now optional.
+4. `Shared_key` field can now be optional with option to choose type of authentication.
+5. Additional fields will be required in sink configuration in Vector's API. See [proposed API](#additional-configuration-fields-for-the-azure-logs-ingestion-sink-to-vector-api) above.
+
+#### CCO Utility Secret
+
+After creation of a managed identity using the CCO utility. The following secret is created and can be used for authentication:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: azure-test-secret
+  namespace: openshift-logging
+type: Opaque
+stringData:
+  azure_client_id: 11111111-2222-3333-4444-555555555555
+  azure_tenant_id: 11111111-2222-3333-4444-555555555555
+  azure_region: westus
+  azure_subscription_id: 11111111-2222-3333-4444-555555555555
+  azure_federated_token_file: /path/to/serviceaccount/token
```

Comment:
Similar to what is required for cloudwatch, I believe we need to dictate this path.  I believe we are stating it as a hard requirement of an explicit path and that it CAN NOT BE anything else.  I think that is where we landed in our discussions.

Reply:
Yes in the API proposal, we can use the projected service account token which is mounted differently from the above. This is just showing the credential secret that is created from the CCO utility